src/time.cpp

```diff
@@ -0,0 +1,53 @@
+#include "cgimap/time.hpp"
+#include <boost/date_time/time_parsing.hpp>
+#include <stdexcept>
+#include <sstream>
+#include <ctype.h>
+
+boost::posix_time::ptime parse_time(const std::string &s) {
```

Comment:
I can't remember why I did it the long way before. Perhaps because it was simple and fixed-format enough to not bother with calling some other function. I've made it call a Boost function in 701caa6. I don't know if it's any clearer now, since the Boost function is pretty much undocumented (and calls other parsing functions).

Apart from testing, it only seems to be called to parse changeset `closed_at` times from the database, which are formatted in a known way, so it shouldn't be a problem.